### DISCUSSION TITLE
关于熵池的疑问

### DISCUSSION THREAD

**Discussion by cangkuai** [canokeys/canokey-product#24] [Q&A] [answered]
众所周知随机数的安全性取决于熵池的安全性，在电脑上有足够的真随机数来源，比如鼠标的位置，指令缓冲区的长度，句柄的创建时间等，但是硬件密钥作为一个完全离线的设备我实在想不到能以什么作为随机数源，当然如果不方便透漏那么就打扰了

**Comment by dangfan** [comment, marked as answer]
请参考：https://zh.wikipedia.org/wiki/%E7%A1%AC%E4%BB%B6%E9%9A%8F%E6%9C%BA%E6%95%B0%E7%94%9F%E6%88%90%E5%99%A8

**Comment by z4yx** [comment]
一般是天然的电路噪声，例如用ADC采样电压噪声，用环振产生相位噪声等
